### PR TITLE
Monitor annotation scanner and monitor compositeData.

### PR DESCRIPTION
**Pull request by kevinliuyifeng**
Change-Id: I8c8516b489db0f716641cd65d641239204a49120

### CODE REVIEW

File: jmx/src/main/java/com/lafaspot/jmetrics/common/MonitorScanner.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+     * @throws OpenDataException creating compositetype fail
+     */
+    private void scan() {
+        final Set<Class<?>> monitorClasses = new HashSet<>();
+        final ClassGraph classGraph = new ClassGraph();
+        classGraph.enableAnnotationInfo().ignoreClassVisibility().blacklistLibOrExtJars()
+                .removeTemporaryFilesAfterScan()
+                .overrideClassLoaders(this.getClass().getClassLoader()).ignoreParentClassLoaders();
+        if (whitelistedPackagesList.size() > 0) {
+            classGraph.whitelistPackages(whitelistedPackagesList.toArray(new String[0]));
```
Comment (nvsnvikram) [inline jmx/src/main/java/com/lafaspot/jmetrics/common/MonitorScanner.java]: Can you tell me if this works?

File: jmx/src/main/java/com/lafaspot/jmetrics/common/MonitorScanner.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        }
+        if (blacklistedPackagesList.size() > 0) {
+            classGraph.blacklistPackages(blacklistedPackagesList.toArray(new String[0]));
```
Comment (nvsnvikram) [inline jmx/src/main/java/com/lafaspot/jmetrics/common/MonitorScanner.java]: Same here.

File: jmx/src/main/java/com/lafaspot/jmetrics/common/MonitorScanner.java [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+            monitorKeyDescription[j] = metricClassComposite.getCompositeType().getDescription();
+            monitorType[j] = metricClassComposite.getCompositeType();
+        }
+
+        try {
+            final CompositeType allMonitorCompositeType = new CompositeType("Monitor Metric", "Monitor Metric Info", monitorKey,
+                    monitorKeyDescription, monitorType);
+            allMonitorCompositeData = new CompositeDataSupport(allMonitorCompositeType, monitorKey, metricClassComposites);
+        } catch (final OpenDataException ox) {
+            logger.error("Creating CompositeData failed", ox);
```
Comment (nvsnvikram) [inline jmx/src/main/java/com/lafaspot/jmetrics/common/MonitorScanner.java]: What happens on jmx if we fail here?

Comment (kevinliuyifeng) [reply jmx/src/main/java/com/lafaspot/jmetrics/common/MonitorScanner.java]: Then scan failed and wait for next scan in 1 min

File: jmx/src/main/java/com/lafaspot/jmetrics/common/MonitorScanner.java (hunk 1 above)
Comment (kevinliuyifeng) [reply jmx/src/main/java/com/lafaspot/jmetrics/common/MonitorScanner.java]: It works.

File: jmx/src/main/java/com/lafaspot/jmetrics/common/MonitorScanner.java (hunk 2 above)
Comment (kevinliuyifeng) [reply jmx/src/main/java/com/lafaspot/jmetrics/common/MonitorScanner.java]: It works

**Pull request closed without merging**